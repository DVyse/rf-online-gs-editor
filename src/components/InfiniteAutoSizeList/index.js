/**
 *
 * InfiniteAutoSizeList
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { InfiniteLoader, AutoSizer, List } from 'react-virtualized';
// import styled from 'styled-components';

/* eslint-disable react/prefer-stateless-function */
class InfiniteAutoSizeList extends React.PureComponent {
  constructor(props) {
    super(props);
    this.forceUpdateGrid = () =>
      this.list ? this.list.forceUpdateGrid() : false;
  }

  render() {
    const {
      rowCount,
      isRowLoaded,
      loadMoreRows,
      rowHeight,
      rowRenderer,
    } = this.props;

    return (
      <InfiniteLoader
        rowCount={rowCount}
        isRowLoaded={isRowLoaded}
        loadMoreRows={loadMoreRows}
      >
        {({ onRowsRendered, registerChild }) => (
          <AutoSizer>
            {({ height, width }) => (
              <List
                ref={list => {
                  this.list = list;
                  return registerChild(this.list);
                }}
                width={width}
                height={height}
                rowCount={rowCount}
                rowHeight={rowHeight}
                onRowsRendered={onRowsRendered}
                rowRenderer={rowRenderer}
              />
            )}
          </AutoSizer>
        )}
      </InfiniteLoader>
    );
  }
}

InfiniteAutoSizeList.propTypes = {
  rowCount: PropTypes.number.isRequired,
  rowHeight: PropTypes.number.isRequired,
  isRowLoaded: PropTypes.func.isRequired,
  loadMoreRows: PropTypes.func.isRequired,
  rowRenderer: PropTypes.func.isRequired,
};

export default InfiniteAutoSizeList;
