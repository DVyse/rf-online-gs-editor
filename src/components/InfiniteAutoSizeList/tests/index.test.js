import React from 'react';
import { shallow } from 'enzyme';

import InfiniteAutoSizeList from '../index';

const isRowLoaded = () => false;
const loadMoreRows = () => Promise.resolve();
const rowRenderer = () => <div />;

describe('<InfiniteAutoSizeList />', () => {
  it('Should exist', () => {
    const wrapper = shallow(
      <InfiniteAutoSizeList
        rowCount={0}
        rowHeight={100}
        isRowLoaded={isRowLoaded}
        loadMoreRows={loadMoreRows}
        rowRenderer={rowRenderer}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
