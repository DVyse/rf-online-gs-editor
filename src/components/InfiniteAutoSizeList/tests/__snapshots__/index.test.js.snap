// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfiniteAutoSizeList /> Should exist 1`] = `
<InfiniteLoader
  isRowLoaded={[Function]}
  loadMoreRows={[Function]}
  minimumBatchSize={10}
  rowCount={0}
  threshold={15}
>
  <Component />
</InfiniteLoader>
`;
